src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -2077,3 +2146,278 @@ std::string Syscollector::query(const std::string& jsonQuery)
+    }
+
+    if (!m_browserExtensions && hasDataInTable(BROWSER_EXTENSIONS_TABLE))
+    {
+        m_disabledCollectorsIndicesWithData.push_back(SYSCOLLECTOR_SYNC_INDEX_BROWSER_EXTENSIONS);
+    }
+
+    if (!m_network)
+    {
+        if (hasDataInTable(NET_IFACE_TABLE))
+        {
+            m_disabledCollectorsIndicesWithData.push_back(SYSCOLLECTOR_SYNC_INDEX_INTERFACES);
+        }
+
+        if (hasDataInTable(NET_PROTOCOL_TABLE))
+        {
+            m_disabledCollectorsIndicesWithData.push_back(SYSCOLLECTOR_SYNC_INDEX_PROTOCOLS);
+        }
+
+        if (hasDataInTable(NET_ADDRESS_TABLE))
+        {
+            m_disabledCollectorsIndicesWithData.push_back(SYSCOLLECTOR_SYNC_INDEX_NETWORKS);
+        }
+    }
+
+    if (!m_disabledCollectorsIndicesWithData.empty() && m_logFunction)
+    {
+        std::string indices;
+
+        for (const auto& idx : m_disabledCollectorsIndicesWithData)
+        {
+            if (!indices.empty())
+            {
+                indices += ", ";
+            }
+
+            indices += idx;
+        }
+
+        m_logFunction(LOG_INFO, "Disabled collectors indices with data detected: " + indices);
+    }
+}
+
+bool Syscollector::notifyDisableCollectorsDataClean()
+{
+    if (m_disabledCollectorsIndicesWithData.empty())
+    {
+        if (m_logFunction)
+        {
+            m_logFunction(LOG_DEBUG, "No disabled collectors indices with data to notify for cleanup");
+        }
+
+        return true;
+    }
+
+    if (!m_spSyncProtocol)
+    {
+        if (m_logFunction)
+        {
+            m_logFunction(LOG_ERROR, "Sync protocol not initialized, cannot notify data clean");
+        }
+
+        return false;
+    }
+
+    if (m_logFunction)
+    {
+        std::string indices;
+
+        for (const auto& idx : m_disabledCollectorsIndicesWithData)
+        {
+            if (!indices.empty())
+            {
+                indices += ", ";
+            }
+
+            indices += idx;
+        }
+
+        m_logFunction(LOG_INFO, "Notifying DataClean for disabled collectors indices: " + indices);
+    }
+
+    return m_spSyncProtocol->notifyDataClean(m_disabledCollectorsIndicesWithData);
+}
+
+void Syscollector::deleteDisableCollectorsData()
+{
+    if (m_disabledCollectorsIndicesWithData.empty())
+    {
+        if (m_logFunction)
+        {
+            m_logFunction(LOG_DEBUG, "No disabled collectors indices with data to delete");
+        }
+
+        return;
+    }
+
+    if (m_logFunction)
+    {
+        std::string indices;
+
+        for (const auto& idx : m_disabledCollectorsIndicesWithData)
+        {
+            if (!indices.empty())
+            {
+                indices += ", ";
+            }
+
+            indices += idx;
+        }
+
+        m_logFunction(LOG_INFO, "Deleting data for disabled collectors indices: " + indices);
+    }
+
+    clearTablesForIndices(m_disabledCollectorsIndicesWithData);
```

Comment:
If `m_allCollectorsDisabled` is true, we can delete the `local.db` and `syscollector_sync.db` databases, just like it’s done in FIM and SCA.

Reply:
Yes, please apply this change.